### DISCUSSION TITLE
Add persistent volume to running cluster

### DISCUSSION THREAD

**Discussion by CoMaDaniel** [strimzi/strimzi-kafka-operator#10808] [Q&A] [answered]
Hello.

May be a duplicate, please excuse me if so.

I have a running Kafka cluster, 3 replicas, with no persistent volumes attached.
I would like to add persistent volumes to the replicas while the cluster is running.

Would it be possible to do so just by adding the volume in the helm chart? Will the data get copied from the ephemeral storage to the persistent one during the rolling update of the brokers?
strimzi version: 0.40.0
kafka version: 3.7.0

Thank you!

**Comment by scholzj** [comment]
You should probably start by sharing your configuration as that defines what can be done with it. Strimzi does not have any Helm chart for deploying the Kafka cluster, only to deploy the operator. So you surely cannot do it there.

**Reply by CoMaDaniel** [reply]
Hello @scholzj .

Thanks for the reply.

Indeed, I deploy the strimzi helm chart and have created my own cart for the kafka cluster.
My question was changing from ephemeral storage to persistent can be done in an active cluster, and the old data would be copied to the persistent storage (when the ISR Catch-Up starts).
Here is my before and after snippet of the Kafka Node pool template:
Before:

apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaNodePool
metadata:
  name: test-dual
  labels:
    strimzi.io/cluster: test
spec:
  replicas: 3
  roles:
    - controller
    - broker
  storage:
    type: ephemeral


After:

apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaNodePool
metadata:
  name: test-dual
  labels:
    strimzi.io/cluster: test
spec:
  replicas: 3
  roles:
    - controller
    - broker
  storage:
    type: jbod
    volumes:
      - id: 0
        type: persistent-claim
        size: 1Gi
        class: default

Thank you!

**Reply by scholzj** [reply, marked as answer]
You should format the code otherwise it is not readable.

But no, as you do not seem to use the JBOD storage (as our docs strongly recommend), you cannot change it easily (+ given Apache KAfka still lacks support for scaling controllers in KRaft, you cannot use the KafkaNodePools for it easily either).
